[common] PONCA_DEBUG macro fails on MacOS and Windows

Using PONCA_DEBUG_ASSERT or PONCA_ASSERT leads to errors on the CI for MacOS and Windows :
https://github.com/poncateam/ponca/actions/runs/24565554206/job/71824714397?pr=301

- On Windows  :
```sh
[13/100] Building CXX object tests\src\CMakeFiles\gls_sphere_der.dir\gls_sphere_der.cpp.obj
FAILED: tests/src/CMakeFiles/gls_sphere_der.dir/gls_sphere_der.cpp.obj 
C:\PROGRA~1\MICROS~2\2022\ENTERP~1\VC\Tools\MSVC\1444~1.352\bin\Hostx64\x64\cl.exe  /nologo /TP  -ID:\a\ponca\ponca\build\tests\src -ID:\a\ponca\ponca\src\tests\src -ID:\a\ponca\ponca\src -ID:\a\ponca\ponca\src\external\eigen /DWIN32 /D_WINDOWS /W3 /GR /EHsc /MD /O2 /Ob2 /DNDEBUG -std:c++20 /bigobj -openmp /showIncludes /Fotests\src\CMakeFiles\gls_sphere_der.dir\gls_sphere_der.cpp.obj /Fdtests\src\CMakeFiles\gls_sphere_der.dir\ /FS -c D:\a\ponca\ponca\src\tests\src\gls_sphere_der.cpp
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(159): error C2760: syntax error: 'volatile' was unexpected here; expected 'expression-statement'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(159): error C2760: syntax error: 'volatile' was unexpected here; expected 'expression'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(159): error C3878: syntax error: unexpected token 'volatile' following 'expression'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(159): error C2760: syntax error: ')' was unexpected here; expected ';'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(159): error C3878: syntax error: unexpected token ')' following 'expression-statement'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(159): error C3878: syntax error: unexpected token ')' following 'statement'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(159): error C3878: syntax error: unexpected token ')' following 'statement-seq'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(159): note: missing one of:  '}' ?
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(159): error C2760: syntax error: ')' was unexpected here; expected '}'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(159): error C3878: syntax error: unexpected token ')' following 'compound-statement'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(159): error C2760: syntax error: ')' was unexpected here; expected 'while'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(159): error C2760: syntax error: ')' was unexpected here; expected '('
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(159): error C2760: syntax error: ')' was unexpected here; expected 'expression'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(159): error C2760: syntax error: '}' was unexpected here; expected 'while'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(159): error C2760: syntax error: '}' was unexpected here; expected '('
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(159): error C2760: syntax error: '}' was unexpected here; expected 'expression'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(159): error C2760: syntax error: '}' was unexpected here; expected ')'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(159): error C2760: syntax error: '}' was unexpected here; expected ';'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(166): error C2760: syntax error: 'volatile' was unexpected here; expected 'expression-statement'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(166): error C2760: syntax error: 'volatile' was unexpected here; expected 'expression'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(166): error C3878: syntax error: unexpected token 'volatile' following 'expression'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(166): error C2760: syntax error: ')' was unexpected here; expected ';'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(166): error C3878: syntax error: unexpected token ')' following 'expression-statement'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(166): error C3878: syntax error: unexpected token ')' following 'statement'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(166): error C3878: syntax error: unexpected token ')' following 'statement-seq'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(166): note: missing one of:  '}' ?
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(166): error C2760: syntax error: ')' was unexpected here; expected '}'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(166): error C3878: syntax error: unexpected token ')' following 'compound-statement'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(166): error C2760: syntax error: ')' was unexpected here; expected 'while'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(166): error C2760: syntax error: ')' was unexpected here; expected '('
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(166): error C2760: syntax error: ')' was unexpected here; expected 'expression'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(166): error C2760: syntax error: '}' was unexpected here; expected 'while'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(166): error C2760: syntax error: '}' was unexpected here; expected '('
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(166): error C2760: syntax error: '}' was unexpected here; expected 'expression'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(166): error C2760: syntax error: '}' was unexpected here; expected ')'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(166): error C2760: syntax error: '}' was unexpected here; expected ';'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(172): error C2760: syntax error: 'volatile' was unexpected here; expected 'expression-statement'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(172): error C2760: syntax error: 'volatile' was unexpected here; expected 'expression'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(172): error C3878: syntax error: unexpected token 'volatile' following 'expression'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(172): error C2760: syntax error: ')' was unexpected here; expected ';'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(172): error C3878: syntax error: unexpected token ')' following 'expression-statement'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(172): error C3878: syntax error: unexpected token ')' following 'statement'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(172): error C3878: syntax error: unexpected token ')' following 'statement-seq'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(172): note: missing one of:  '}' ?
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(172): error C2760: syntax error: ')' was unexpected here; expected '}'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(172): error C3878: syntax error: unexpected token ')' following 'compound-statement'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(172): error C2760: syntax error: ')' was unexpected here; expected 'while'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(172): error C2760: syntax error: ')' was unexpected here; expected '('
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(172): error C2760: syntax error: ')' was unexpected here; expected 'expression'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(172): error C2760: syntax error: '}' was unexpected here; expected 'while'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(172): error C2760: syntax error: '}' was unexpected here; expected '('
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(172): error C2760: syntax error: '}' was unexpected here; expected 'expression'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(172): error C2760: syntax error: '}' was unexpected here; expected ')'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(172): error C2760: syntax error: '}' was unexpected here; expected ';'
D:\a\ponca\ponca\src\Ponca\src\SpatialPartitioning\../Common/Containers/limitedPriorityQueue.h(185): error C2760: syntax error: 'volatile' was unexpected here; expected 'expression-statement'
```

- On MacOS : 
```sh

/Users/runner/work/ponca/ponca/src/Ponca/src/SpatialPartitioning/KdTree/Query/../../../Common/Containers/limitedPriorityQueue.h:173:9: error: unrecognized instruction mnemonic, did you mean: bit, cnt, hint, ins, not?
  173 |         PONCA_ASSERT(capacity <= N);
      |         ^
/Users/runner/work/ponca/ponca/src/Ponca/src/SpatialPartitioning/KdTree/../../Common/Assert.h:21:9: note: expanded from macro 'PONCA_ASSERT'
   21 |         PONCA_ERROR_MSG(PONCA_STR(EXPR)); \
      |         ^
/Users/runner/work/ponca/ponca/src/Ponca/src/SpatialPartitioning/KdTree/../../Common/Assert.h:14:5: note: expanded from macro 'PONCA_ERROR_MSG'
   14 |     PONCA_CRASH;             \
      |     ^
/Users/runner/work/ponca/ponca/src/Ponca/src/SpatialPartitioning/KdTree/./../../Common/Macro.h:19:18: note: expanded from macro 'PONCA_CRASH'
   19 |     asm volatile("int $3"); \
      |                  ^
<inline asm>:1:2: note: instantiated into assembly here
    1 |         int $3
      |         ^
```

This is provoked by the following line, which is compiler dependent : 

https://github.com/poncateam/ponca/blob/73247abfe24d29406a95aee1d4dfa2d34da85d4c/Ponca/src/Common/Macro.h#L17-L20

We should replace it with something like `__builtin_trap`

--- a/CHANGELOG
+++ b/CHANGELOG
@@ -7,6 +7,7 @@ This release will introduce a refactoring of the requirement library using cpp20
 
 - Bug-fixes and code improvements
     - [Fitting] Fix warnings introduced when bumping to cxx20 (#303)
+    - [common] Fix PONCA_ASSERT macro on MSVC and clang (#309)
 - Cmake
     - [all] Eigen is now always for transitive linking (#302)
     - [spatialPartitioning] Fix regression introduced by the CUDA kdtree update (#306)

--- a/Ponca/src/Common/Macro.h
+++ b/Ponca/src/Common/Macro.h
@@ -14,9 +14,18 @@
 #define PONCA_XSTR(S) #S
 #define PONCA_STR(S) PONCA_XSTR(S)
 
-#define PONCA_CRASH         \
-    PONCA_MACRO_START       \
-    asm volatile("int $3"); \
+#if defined(_MSC_VER)
+#    define PONCA_ABORT __debugbreak();
+#elif defined(__clang__) || defined(__GNUC__)
+#    define PONCA_ABORT __builtin_trap();
+#else
+#    include <cstdlib>
+#    define PONCA_ABORT std::abort();
+#endif
+
+#define PONCA_CRASH   \
+    PONCA_MACRO_START \
+    PONCA_ABORT       \
     PONCA_MACRO_END
 
 #define PONCA_PRINT_ERROR(MSG)                                       \

